### DISCUSSION TITLE
How to find the reason for missing packages?

### DISCUSSION THREAD

**Discussion by probonopd** [helloSystem/hello#384] [Q&A]
From https://github.com/helloSystem/ISO/issues/141#issuecomment-1256958678:

FreeBSD packages can be missing from one day to the next. Each time this happens, we can't build helloSystem ISOs, which is very annoying. Today we can't build the FreeBSD 14 based ISO because `pkg-static: No packages available to install matching 'virtualbox-ose-additions' have been found in the repositories`.

Let's see if I can figure it out:

* Go to https://pkg-status.freebsd.org/?type=package&all=1
* Search `main-amd64` (which is FreeBSD 14)
* It says `beefy18`, which is the machine on which FreeBSD 14 packages get built for amd64
* Go to its latest build, which is http://beefy18.nyi.freebsd.org/build.html?mastername=main-amd64-default&build=pff8f3a1b75aa_s1b602f641a
* Search for `virtualbox-ose-additions` in all search boxes on that page; it is found under "Queued ports"
* What does this tell us now? Why is this package missing today?

**Comment by grahamperrin** [comment]
> … latest build, which is http://beefy18.nyi.freebsd.org/build.html?mastername=main-amd64-default&build=pff8f3a1b75aa_s1b602f641a …

![image](https://user-images.githubusercontent.com/192271/192101912-b7e59a58-ec55-46ac-8f47-930e0c948b77.png)

Open a log, check the date. 

> FreeBSD 14 

See `FreeBSD:14:amd64` under <https://github.com/helloSystem/hello/discussions/382#discussioncomment-3671074>; NB the reason for the different steps.

**Reply by probonopd** [reply]
So, were you able to find out the reason why it is missing?

(That user interface is very confusing for me.)

**Reply by grahamperrin** [reply]
At <https://pkg-status.freebsd.org/?type=package&all=1> (all packages) filtered for _main-amd64_, you probably forgot to **sort**: 

![image](https://user-images.githubusercontent.com/192271/192113522-1af12764-6ce1-4fc6-b1ed-935f715f49b0.png)

– to get the most recent _stopped:done:_ build at, or near, the top of the column. 

Next: 

![image](https://user-images.githubusercontent.com/192271/192113601-8e7edb02-69d6-4ed1-9676-91d2fc046de2.png)

– leads to <http://beefy18.nyi.freebsd.org/build.html?mastername=main-amd64-default&build=p47ff75a1891e_s2c2ef670a7>, here's the relevant log: 

![image](https://user-images.githubusercontent.com/192271/192113730-965ab67b-24ea-43b1-90b3-2b34dbb7409b.png)

* <http://beefy18.nyi.freebsd.org/data/main-amd64-default/p47ff75a1891e_s2c2ef670a7/logs/errors/virtualbox-ose-additions-6.1.36_1.log>

**Comment by grahamperrin** [comment]
From <https://github.com/helloSystem/ISO/issues/141#issuecomment-778729491> (2021-02-14): 

> … The key to avoiding disappointment will be knowledge of fallout of required packages. …

From **Contributing to FreeBSD**: [Finding and fixing a broken port](https://docs.freebsd.org/en/articles/contributing/#fix-broken)

> … [PortsFallout](https://portsfallout.com/) shows port issues gathered from the FreeBSD package building. …

### Seeking `emulators/virtualbox-ose-additions$`

![image](https://user-images.githubusercontent.com/192271/192114307-249b86c9-9e97-43fe-9ee9-bee9e3e071a9.png)

Result: <https://portsfallout.com/fallout?port=emulators%2Fvirtualbox-ose-additions%24&maintainer=&env=&category=&flavor=>

**Reply by probonopd** [reply]
Cool. So `emulators/virtualbox-ose-additions` is failing to build on FreeBSD 14 but not 13:

```
/wrkdirs/usr/ports/emulators/virtualbox-ose-additions/work/VirtualBox-6.1.36/src/VBox/Additions/freebsd/vboxvfs/vboxvfs_vnops.c:1304:22: error: use of undeclared identifier 'SAVENAME'; did you mean 'RENAME'?
                                cnp->cn_flags |= SAVENAME;
                                                 ^~~~~~~~
                                                 RENAME
/usr/src/sys/sys/namei.h:44:40: note: 'RENAME' declared here
enum nameiop { LOOKUP, CREATE, DELETE, RENAME };
                                       ^
3 warnings and 1 error generated.
kmk: *** [/wrkdirs/usr/ports/emulators/virtualbox-ose-additions/work/VirtualBox-6.1.36/out/freebsd.amd64/release/obj/vboxvfs/vboxvfs_vnops.o] Error 1
```

**Reply by probonopd** [reply]
PortsFallout is cool, but it doesn't explain why `print/hplip` is still missing for FreeBSD 14:

https://portsfallout.com/port?port=%5Eprint/hplip$

**Reply by grahamperrin** [reply]
https://github.com/FreshPorts/freshports/issues/384#issuecomment-1257274629

**Reply by grahamperrin** [reply]
https://github.com/helloSystem/ISO/issues/141#issuecomment-1279952635

> Thanks @grahamperrin for having submitted a bug for the missing 'virtualbox-ose-additions' upstream:
> 
> [bugs.freebsd.org/bugzilla/show_bug.cgi?id=267079](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=267079)

Certainly useful for tracking, however please know that maintainers were already aware of the failures to build. 

At **[Dashboard - FreeBSD pkg-fallout](https://portsfallout.com/)** the _maintainer_ shortlist currently includes: 

* [vbox@freebsd.org](https://portsfallout.com/fallout?maintainer=vbox%40FreeBSD.org)

– leading to the report for each failure, and more.

**Comment by grahamperrin** [comment]
### [www/falkon](https://www.freshports.org/www/falkon/)

#### falkon-qtonly

<https://github.com/helloSystem/hello/issues/161#issuecomment-1596238688> (2023-06-18 20:08 GMT+1)

> … quarterly …

<https://github.com/helloSystem/hello/issues/161#issuecomment-1596268487>

> Seems to be qt6-webengine failing.
> 
> [pkg-status.freebsd.org/builds/default:default:131amd64:8dc4a71c010e:beefy16](https://pkg-status.freebsd.org/builds/default:default:131amd64:8dc4a71c010e:beefy16)

_beefy16_ was latest – _default_. Not quarterly.

----

**Quarterly**

<https://pkg-status.freebsd.org/?type=package&all=1>, suitably filtered and sorted: 

![image](https://github.com/helloSystem/hello/assets/192271/f64c99fe-034e-4180-83d5-13f48ea05c80)

<https://pkg-status.freebsd.org/builds?server=beefy14&all=1>, suitably filtered and sorted: 

![image](https://github.com/helloSystem/hello/assets/192271/329ff80d-4bb4-4022-a7ef-19c3c545284e)

<http://beefy14.nyi.freebsd.org/build.html?mastername=131amd64-quarterly&build=098d7e1a1744> the reason for the skip of falkon-qtonly-22.12.3 was failure of qt5-webengine-5.15.8.

**Reply by grahamperrin** [reply]
<http://beefy14.nyi.freebsd.org/build.html?mastername=131amd64-quarterly&build=15aa9ea6ac29>

* <http://beefy14.nyi.freebsd.org/data/131amd64-quarterly/15aa9ea6ac29/logs/falkon-qtonly-22.12.3.log> build of falkon-qtonly-22.12.3 started at **2023-06-19 11:59:27** UTC, succeeded in less than nine minutes.

**Comment by grahamperrin** [comment]
### [www/qt5-webengine](https://www.freshports.org/www/qt5-webengine/)

<http://beefy14.nyi.freebsd.org/data/131amd64-quarterly/098d7e1a1744/logs/errors/qt5-webengine-5.15.8.log> 

```text
=>> Building www/qt5-webengine
build started at Thu Jun 15 08:07:17 UTC 2023
port directory: /usr/ports/www/qt5-webengine
package name: qt5-webengine-5.15.8
building for: FreeBSD 131amd64-quarterly-job-08 13.1-RELEASE-p7 FreeBSD 13.1-RELEASE-p7 amd64
maintained by: kde@FreeBSD.org
Makefile ident: 
Poudriere version: 3.2.8-23-ga7f8d188
Host OSVERSION: 1400089
Jail OSVERSION: 1301000
Job Id: 08

…
FAILED: obj/extensions/browser/api/usb/usb/usb_api.o 
…
FAILED: obj/extensions/browser/api/usb/usb/usb_device_manager.o 
…
ninja: build stopped: subcommand failed.
*** [run_ninja] Error code 1

make[4]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src/core
1 error

make[4]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src/core
*** [sub-gn_run-pro-make_first] Error code 2

make[3]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src/core
1 error

make[3]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src/core
*** [sub-core-make_first] Error code 2

make[2]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src
1 error

make[2]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build/src
*** [sub-src-make_first] Error code 2

make[1]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build
1 error

make[1]: stopped in /wrkdirs/usr/ports/www/qt5-webengine/work/.build
===> Compilation failed unexpectedly.
Try to set MAKE_JOBS_UNSAFE=yes and rebuild before reporting the failure to
the maintainer.
*** Error code 1

Stop.
make: stopped in /usr/ports/www/qt5-webengine
=>> Cleaning up wrkdir
===>  Cleaning for qt5-webengine-5.15.8
build of www/qt5-webengine | qt5-webengine-5.15.8 ended at Thu Jun 15 14:42:19 UTC 2023
build time: 06:35:02
!!! build failure encountered !!!
```

**Reply by grahamperrin** [reply]
<http://beefy14.nyi.freebsd.org/build.html?mastername=131amd64-quarterly&build=15aa9ea6ac29>

* <http://beefy14.nyi.freebsd.org/data/131amd64-quarterly/15aa9ea6ac29/logs/qt5-webengine-5.15.8.log> (88.7 MiB) build of qt5-webengine started at **2023-06-18 02:50:24** UTC, succeeded in nine hours – ended at 11:50:48 UTC.

**Comment by pafkakummu** [comment]
thank you for suggestions! useful information!
